2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,372 @@
     }
 
     @Test
     @Order(2)
-    void size() {
-        integerQueue.add(98);
-        integerQueue.add(9);
-        integerQueue.add(5);
-        integerQueue.add(6);
+    void chekFieldsNameInNodeClass() {
+        Class<?> innerClass = getInnerStaticNodeClass();
+        boolean hasElementField = hasField(innerClass, ELEMENT_FIELD);
+        boolean hasNodeField = hasField(innerClass, NODE_FIELD);
 
-        assertEquals(4, integerQueue.size());
+        assertThat(hasElementField).isTrue();
+        assertThat(hasNodeField).isTrue();
     }
 
     @Test
     @Order(3)
-    void addElementIntoEmptyQueue() {
-        integerQueue.add(1312);
+    void checkFieldsInQueueCLass() {
+        Class<?> baseClass = this.integerQueue.getClass();
+        boolean hasHeadField = hasField(baseClass, HEAD_FIELD);
+        boolean hasSizeFiled = hasField(baseClass, SIZE_FIELD);
+        boolean hasTailFiled = hasField(baseClass, TAIL_FIELD);
 
-        assertEquals(1, integerQueue.size());
-        assertEquals(1312, integerQueue.poll().intValue());
+        assertThat(hasHeadField).isTrue();
+        assertThat(hasSizeFiled).isTrue();
+        assertThat(hasTailFiled).isTrue();
     }
 
     @Test
     @Order(4)
-    void addElement() {
-        integerQueue.add(324);
-        integerQueue.add(23);
-        integerQueue.add(5);
+    void addFillsQueueWhenItIsEmpty() {
+        integerQueue.add(1);
+        integerQueue.add(228);
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = (Integer) pollElementFromQueue();
+        Integer secondElement = (Integer) pollElementFromQueue();
 
-        assertEquals(3, integerQueue.size());
-        assertEquals(324, integerQueue.poll().intValue());
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(1);
+        assertThat(secondElement).isEqualTo(228);
     }
 
     @Test
     @Order(5)
-    void pollElementFromEmptyQueue() {
-        assertNull(integerQueue.poll());
+    void addFillsQueueWhenItIsNotEmpty() {
+        addIntElementToQueue(12);
+        integerQueue.add(111);
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = (Integer) pollElementFromQueue();
+        Integer secondElement = (Integer) pollElementFromQueue();
+
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(12);
+        assertThat(secondElement).isEqualTo(111);
     }
 
     @Test
     @Order(6)
-    void pollElement() {
-        integerQueue.add(33);
-        integerQueue.add(123);
-        integerQueue.add(222);
-        integerQueue.add(444);
-
-        integerQueue.poll(); // should poll 33
+    void addIncreasesQueueSize() {
+        integerQueue.add(1);
+        integerQueue.add(2);
+        int size = this.integerQueue.size();
```

Comment:
You should use reflection method here to get size. If size method is not implemented the test doesn't pass now.